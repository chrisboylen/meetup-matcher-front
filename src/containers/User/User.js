import React from 'react';
import { PropTypes } from 'prop-types';
import { logOutFirebase } from '../../firebase/auth';
import { logoutUser } from '../../actions';
import { connect } from 'react-redux';
import Header from '../Header/Header';
import { Questions } from '../Questions/Questions';


export const User = ({ user, }) => {
  return(
    <div className="user">
      <Header />
      <Questions />
    </div>
  )
};

User.propTypes = {
  user: PropTypes.object
};

export const mapStateToProps = ({ user }) => ({ user });

export const mapDispatchToProps = (dispatch) => ({
  logoutUser: () => dispatch(logoutUser())
});

export default connect(mapStateToProps, mapDispatchToProps)(User);