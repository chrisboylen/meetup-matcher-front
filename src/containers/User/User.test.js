import React from 'react';
import { shallow } from 'enzyme';
import { User, mapStateToProps, mapDispatchToProps } from './User';

describe('User Component', () => {
  it('should match snapshot when user has questions', () => {
    const mockUser = {questions: 'yes'}
    const wrapper = shallow(<User user={mockUser} />);

    expect(wrapper).toMatchSnapshot()
  });
  it('should match snapshot when user has no questions', () => {
    const mockUser = {questions: ''}
    const wrapper = shallow(<User user={mockUser} />);

    expect(wrapper).toMatchSnapshot()
  });
});