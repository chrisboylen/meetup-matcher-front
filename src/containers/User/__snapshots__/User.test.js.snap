// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Component should match snapshot when user has no questions 1`] = `
<div
  className="user"
>
  <withRouter(Connect(Header)) />
  <withRouter(Connect(Questions)) />
</div>
`;

exports[`User Component should match snapshot when user has questions 1`] = `
<div
  className="user"
>
  <withRouter(Connect(Header)) />
  <withRouter(Connect(CitySearch)) />
</div>
`;
